### PR TITLE
feat: Updated cam & ped natives

### PR DESCRIPTION
**Pull request by JajarGaming**
Switched native hash thing to the name

### CODE REVIEW

File: PLAYER/RestorePlayerStamina.md [hunk 1]
```diff
@@ -7,7 +7,22 @@ ns: PLAYER
 // 0xA352C1B864CAFD33 0x62A93608
 void RESTORE_PLAYER_STAMINA(Player player, float p1);
 ```
+```
```
Comment (4mmonium) [inline PLAYER/RestorePlayerStamina.md:10]: Do you think you could remove this entire code block, add a proper description and provide a working example in Lua?

This code [example](https://github.com/citizenfx/natives/blob/master/HUD/EndTextCommandThefeedPostAward.md?plain=1#L36) shows how it should be written by the way.

You can also check out the contribution guide [here](https://github.com/citizenfx/natives/blob/master/.github/CONTRIBUTING.md)

**Review by 4mmonium — CHANGES_REQUESTED** [review]
Check over what I commented, your effort into contributing is appreciated, thank you!

**Comment by JajarGaming** [comment]
> Check over what I commented, your effort into contributing is appreciated, thank you!

Saw it! Will do the changes hopefully soon...

**Comment by JajarGaming** [comment]
Haven't tested what the second argument does, if its a multiplier of some sorts or whatever.. Don't have the time to do it currently 🤷🏼‍♂️

File: PLAYER/RestorePlayerStamina.md [hunk 2]
```diff
@@ -7,9 +7,21 @@ ns: PLAYER
 // 0xA352C1B864CAFD33 0x62A93608
 void RESTORE_PLAYER_STAMINA(Player player, float p1);
 ```
+Regenerate a players stamina. Unsure if p1 is some sort of multiplier or regen speed, need to look into this 🤔.
```
Comment (4mmonium) [inline PLAYER/RestorePlayerStamina.md:10]: I just ran some tests and based on my research, `p1` seems to be a percentage that ranges from 0.0 to 1.0 (1.0 being 100%) and it seems to add a percentage of stamina to the player.

If the player's stamina is 10.0 and you restore 10% stamina (0.10) you get 20% stamina on the player.

This is the code I used to run my tests:
```lua
-- Inside a while loop
Citizen.Trace("Stamina: " .. GetPlayerStamina(PlayerId()))

-- Command
RegisterCommand("rs", function(source, args, rawCommand)
  RestorePlayerStamina(PlayerId(), 0.1)
end, false)
```

The game's native method (my name guesses may be wrong):

```c
void __fastcall restoreStamina(__int64 ped, float percentageToRestore)
{
  float currentStaminaValue; // xmm2_4
  float newStaminaValue; // xmm0_4

  if ( percentageToRestore >= 0.0 && percentageToRestore <= 100.0 )// This is wrong, should be <= 1.0
  {
    currentStaminaValue = *(float *)(ped + 0xBB4);
    newStaminaValue = (float)(currentStaminaValue * percentageToRestore) + *(float *)(ped + 0xBB0);
    *(float *)(ped + 0xBB0) = newStaminaValue;
    if ( newStaminaValue >= currentStaminaValue )
      *(float *)(ped + 2992) = currentStaminaValue;
  }
}
```

Comment (JajarGaming) [reply PLAYER/RestorePlayerStamina.md:10]: I'll update the documentation 👍🏼

File: PED/GetPedDrawableVariation.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ Ids
 4 - Legs
 5 - Hands
 6 - Foot
-7 - ------
+7 - Scarfs/Neck Accessories
 8 - Accessories 1
 9 - Accessories 2
 10- Decals
```
Comment (4mmonium) [inline PED/GetPedDrawableVariation.md:23]: Huh, there's one more apparently, `11` seems to be `JBIB`, which is used for torso suits/jackets

```c
if ( drawableVariationIdx > 6 )
    {
      switch ( drawableVariationIdx )
      {
        case 7:
          v7 = "TEETH";
          break;
        case 8:
          v7 = "ACCESSORIES";
          break;
        case 9:
          v7 = "TASK";
          break;
        case 10:
          v7 = "DECL";
          break;
        default: // Most likely 11
          v7 = "JBIB";
          break;
      }
```

You don't have to add it though, I could add it at a later PR 🙂

**Review by 4mmonium — APPROVED** [review]
Approved, thank you for your contribution! 😃

File: PED/GetPedDrawableVariation.md (hunk 3 above)
Comment (JajarGaming) [reply PED/GetPedDrawableVariation.md:23]: 11 is jackets. Some of the other ones might be wrong as well, I was making a clothing customisation menu and was researching all of the drawables and props. So I have a list of all the correct ones.

Comment (4mmonium) [reply PED/GetPedDrawableVariation.md:23]: > 11 is jackets. Some of the other ones might be wrong as well, I was making a clothing customisation menu and was researching all of the drawables and props. So I have a list of all the correct ones on my computer, won't be available for the weekend tho...

Yeah, I searched for "JBIB" in the rpf files and figured it had something to do with jackets as you are saying. Anyway, that's ok. Have a great weekend!

Comment (JajarGaming) [reply PED/GetPedDrawableVariation.md:23]: Yep. You too 😄

**Pull request merged**